coldfront/core/utils/management/commands/update_state_and_extra_fields.py

```diff
@@ -0,0 +1,88 @@
+'''
+update old requests with new default values for state and extra_fields
+STATE:
+AllocationAdditionRequest, SecureDirRequest, SavioProjectAllocationRequest (Recharge, ICA):
+        'notified': {
+            'status': 'Pending',
+            'timestamp': '', }
+
+EXTRA_FIELDS:
+SavioProjectAllocationRequest (ICA):
+        'course_name': '',
+        'course_department': '',
+        'point_of_contact': '',
+'''
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from coldfront.core.allocation.models import (AllocationAdditionRequest,
+                                              SecureDirRequest)
+from coldfront.core.project.models import SavioProjectAllocationRequest
+from coldfront.core.user.models import UserProfile
+from coldfront.core.allocation.models import AllocationAttribute
+
+from coldfront.core.project.utils_.renewal_utils import \
+    get_current_allowance_year_period
+from coldfront.core.resource.utils_.allowance_utils.computing_allowance import \
+    ComputingAllowance
+from coldfront.core.utils.common import display_time_zone_current_date
+
+class Command(BaseCommand):
+    help = 'Update the state and extra fields for Project Allocation Requests' \
+           ', Secure Directory Requests, and Allocation Addition Requests.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--all', action='store_true',
+            help='Update both state and extra fields')
+        parser.add_argument('--state', action='store_true',
+            help='Update state')
+        parser.add_argument('--extra-fields', action='store_true',
+            help='Update extra fields')
+        parser.add_argument('--dry-run', action='store_true',
```

Comment:
Use `coldfront.core.utils.common.add_argparse_dry_run_argument` for consistency. Note that the name will be `--dry_run`, not `--dry-run`.

Would line 73 have worked given that it was named `dry-run`? Not sure.

Reply:
yeah django converts hyphens to underscores